File: DSA/Ishita-Nigam/Day5/Day5_Q2.java
import java.util.Scanner;

class ListNode {
    int val;
    ListNode next;

    ListNode(int val) {
        this.val = val;
        this.next = null;
    }
}

public class Day5_Q2 {
    public static String compareLists(ListNode head1, ListNode head2) {
        while (head1 != null && head2 != null) {
            if (head1.val != head2.val) {
                return "different";
            }
            head1 = head1.next;
            head2 = head2.next;
        }

        if (head1 == null && head2 == null) {
            return "identical";
        } else {
            return "different";
        }
    }
    public static ListNode createList() {
        Scanner scanner = new Scanner(System.in);
        ListNode head = null, current = null;
        System.out.println("Enter elements of the linked list (enter -1 to stop):");

        while (true) {
            int val = scanner.nextInt();
            if (val == -1) {
                break;
            }

            ListNode newNode = new ListNode(val);

            if (head == null) {
                head = newNode;
                current = head;
            } else {
                current.next = newNode;
                current = current.next;
            }
        }

        return head;
    }

    public static void main(String[] args) {
        System.out.println("Creating first linked list:");
        ListNode head1 = createList();

        System.out.println("Creating second linked list:");
        ListNode head2 = createList();

        // Comparing the two linked lists
        System.out.println("The lists are " + compareLists(head1, head2));
    }
}

File: DSA/saumya/Day5_Q2.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


def compare_lists(head1, head2):
 #       head1: The head node of the first linked list.
  #      head2: The head node of the second linked list.

   # returns "identical" if the lists are the same, otherwise "different".   
    while head1 and head2:
        if head1.data != head2.data:
            return "different"
        head1 = head1.next
        head2 = head2.next

    # If one list is longer than the other, they are different.
    if head1 or head2:
        return "different"

    return "identical"

head1 = Node(1)
head1.next = Node(2)
head1.next.next = Node(3)

head2 = Node(1)
head2.next = Node(2)
head2.next.next = Node(3)

print(compare_lists(head1, head2))   # Output: identical

head3 = Node(1)
head3.next = Node(2)
head3.next.next = Node(4)

print(compare_lists(head1, head3))    # Output: different
